extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalog.java

```diff
@@ -0,0 +1,243 @@
+        }
+        return builder.build();
+    }
+
+    @SuppressWarnings("unused")
+    public List<TableIdentifier> listTables(final Namespace namespace) {
+        // TODO: have this return a Deephaven Table of table identifiers
+        return catalog.listTables(namespace);
+    }
+
+    /**
+     * Read a static snapshot of a table from the Iceberg catalog.
+     *
+     * @param tableIdentifier The table identifier to load
+     * @param snapshotId The snapshot ID to load
+     * @return The loaded table
+     */
+    @SuppressWarnings("unused")
+    public Table readTable(
+            @NotNull final TableIdentifier tableIdentifier,
+            @NotNull final String snapshotId) {
+        return readTableInternal(tableIdentifier, snapshotId, false);
+    }
+
+    /**
+     * Read the latest static snapshot of a table from the Iceberg catalog.
+     *
+     * @param tableIdentifier The table identifier to load
+     * @return The loaded table
+     */
+    @SuppressWarnings("unused")
+    public Table readTable(@NotNull final TableIdentifier tableIdentifier) {
+        return readTableInternal(tableIdentifier, null, false);
+    }
+
+    /**
+     * Subscribe to a table from the Iceberg catalog. Initially the latest snapshot will be loaded, but the output table
+     * will be updated as new snapshots are added to the table.
+     *
+     * @param tableIdentifier The table identifier to load
+     * @return The loaded table
+     */
+    @SuppressWarnings("unused")
+    public Table subscribeTable(@NotNull final TableIdentifier tableIdentifier) {
+        return readTableInternal(tableIdentifier, null, true);
+    }
+
+    private Table readTableInternal(
+            @NotNull final TableIdentifier tableIdentifier,
+            @Nullable final String snapshotId,
+            final boolean isRefreshing) {
+        // Validate that the user is not trying to subscribe to a snapshot.
+        Assert.eqFalse(isRefreshing && (snapshotId != null),
+                "Must not specify a snapshot ID when subscribing to a table.");
+
+        // Load the table from the catalog
+        final org.apache.iceberg.Table table = catalog.loadTable(tableIdentifier);
+
+        final Snapshot snapshot;
+        final Schema schema;
+        // Do we want the latest or a specific snapshot?
+        if (snapshotId == null) {
+            snapshot = table.currentSnapshot();
+            schema = table.schema();
+        } else {
+            // Load the specific snapshot and retrieve the schema for that snapshot
+            snapshot = table.snapshot(snapshotId);
+            schema = table.schemas().get(snapshot.schemaId());
+        }
+
+        // Load the partitioning schema
+        final org.apache.iceberg.PartitionSpec partitionSpec = table.spec();
+
+        // Convert the Iceberg schema to a Deephaven TableDefinition
+        final TableDefinition tableDefinition = IcebergTools.fromSchema(schema, partitionSpec);
```

Comment:
When we have a computed partition column, does it always appear as a partitioning column of the Deephaven table definition, or do we just get the "source" column?
You say the former, which is probably best for DH users.